src/clients/FtpGateway.cc

```diff
@@ -1257,6 +1263,7 @@ Ftp::Gateway::loginFailed()
 
     // add it to the store entry for response....
     entry->replaceHttpReply(newrep);
+    fwd->markStoredReplyAsWhole("loginFailed");
```

Comment:
This fixes the 'entry->completeTruncated()' problem and adding 'WITH_CLIENT' to err_detail. I fixed other cases in FtpGateway as well.